packages/tools/pluggable-widgets-tools/configs/rollup.config.js

```diff
@@ -0,0 +1,278 @@
+import { getBabelInputPlugin, getBabelOutputPlugin } from "@rollup/plugin-babel";
+import commonjs from "@rollup/plugin-commonjs";
+import json from "@rollup/plugin-json";
+import { nodeResolve } from "@rollup/plugin-node-resolve";
+import replace from "@rollup/plugin-replace";
+import typescript from "@rollup/plugin-typescript";
+import url from "@rollup/plugin-url";
+import loadConfigFile from "rollup/dist/loadConfigFile";
+import clear from "rollup-plugin-clear";
+import command from "rollup-plugin-command";
+import livereload from "rollup-plugin-livereload";
+import sass from "rollup-plugin-sass";
+import { terser } from "rollup-plugin-terser";
+import { cp } from "shelljs";
+import { zip } from "zip-a-folder";
+import { widgetTyping } from "./rollup-plugin-widget-typing";
+import {
+    editorConfigEntry,
+    isTypescript,
+    previewEntry,
+    projectPath,
+    sourcePath,
+    widgetEntry,
+    widgetName,
+    widgetPackage,
+    widgetVersion
+} from "./shared";
+
+const outDir = join(sourcePath, "/dist/tmp/widgets/");
+const outWidgetFile = join(widgetPackage.replace(/\./g, "/"), widgetName.toLowerCase(), `${widgetName}.js`);
+const mpkDir = join(sourcePath, "dist", widgetVersion);
+const mpkFile = join(mpkDir, `${widgetPackage}.${widgetName}.mpk`);
+
+export default async args => {
+    const platform = args.configPlatform;
+    const production = Boolean(args.configProduction);
+    if (!["web", "native"].includes(platform)) {
+        throw new Error("Must pass --configPlatform=web|native parameter");
+    }
+    const result = [];
+
+    if (platform === "web") {
+        result.push({
+            input: widgetEntry,
+            output: {
+                format: "amd",
+                file: join(outDir, outWidgetFile),
+                sourcemap: !production ? "inline" : false
+            },
+            external: [/^mendix($|\/)/, "react", "react-dom", "big.js"],
+            plugins: [
+                ...getClientComponentPlugins(),
+                url({ include: imagesAndFonts, limit: 100000 }),
+                sass({ output: true, include: /\.(css|sass|scss)$/ }),
+                alias({
+                    entries: {
+                        "react-hot-loader/root": join(__dirname, "hot")
+                    }
+                }),
+                ...getCommonPlugins({
+                    sourceMaps: !production,
+                    extensions: webExtensions,
+                    transpile: production,
+                    babelConfig: {
+                        presets: [["@babel/preset-env", { targets: { safari: "12" } }]],
+                        allowAllFormats: true
+                    }
+                })
+            ],
+            onwarn
+        });
+    }
+
+    if (platform === "native") {
+        result.push({
+            input: widgetEntry,
+            output: {
+                format: "es",
+                file: join(outDir, outWidgetFile),
+                sourcemap: false
```

Comment:
Dont we want to have a source map for the native like we do for web on line 105 ?

Reply:
Not supported

Reply:
Yep, metro does not support copying/re-mapping sourcemaps => it can have sourcemaps only for the code it transpiles.